webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTable.js

```diff
@@ -39,55 +40,58 @@ ErrorAlerts.propTypes = {
   errors: PropTypes.arrayOf(PropTypes.string).isRequired,
 };
 
-const buildTableRows = (subscriptions, updatedQuantity) =>
-  subscriptions.results.map((subs) => {
-    if (updatedQuantity[subs.id]) {
-      return {
-        ...subs,
-        entitlementsChanged: true,
-        quantity: updatedQuantity[subs.id],
-        availableQuantity: subscriptions.availableQuantities[subs.id],
-      };
-    }
-    return {
-      ...subs,
-      availableQuantity: subscriptions.availableQuantities[subs.id],
-    };
-  });
-
-const buildPools = updatedQuantity =>
-  Object.entries(updatedQuantity)
-    .map(([id, quantity]) => ({
-      id,
-      quantity,
-    }));
-
 class SubscriptionsTable extends Component {
-  static getDerivedStateFromProps(nextProps, prevState) {
-    if (nextProps.subscriptions !== undefined) {
-      return {
-        rows: buildTableRows(
-          nextProps.subscriptions,
-          prevState.updatedQuantity,
-        ),
-      };
-    }
-    return null;
-  }
-
   constructor(props) {
     super(props);
+
     this.state = {
+      rows: undefined,
+      subscriptions: undefined,
+      groupdSubscriptions: undefined,
       updatedQuantity: {},
       editing: false,
-      rows: props.subscriptions.results,
       showUpdateConfirmDialog: false,
       showCancelConfirmDialog: false,
       showErrorDialog: false,
       selectedRows: [],
     };
   }
 
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (
+      nextProps.subscriptions !== undefined &&
+      !isEqual(nextProps.subscriptions, prevState.subscriptions)
+    ) {
+      const groupdSubscriptions = groupSubscriptionsByProductId(nextProps.subscriptions);
+      const rows = buildTableRows(
```

Comment:
theses functions invoked before each render, even if nothing has changed! which leads to a poor performance 
I think it would be much better use memorization helper here

Reply:
Yes, the function run before each render, but it will run the inner logic only if the subscriptions received from redux changed.

Once we will move all the state to redux, we would be able to implement memorized selectors.